### PR TITLE
Configuration: add "Global profile configuration"

### PR DESCRIPTION
**Pull request by jdeniau**
Add documentation for https://github.com/Behat/Behat/pull/1501 and https://github.com/Behat/Behat/pull/1513

### CODE REVIEW

File: user_guide/configuration.rst [hunk 1]
```diff
@@ -111,6 +111,40 @@ All files from the ``imports`` block will be loaded by Behat and merged, in the
...
+Global profile configuration
+----------------------------
+
+You can set some global configuration in your profile configuration:
+
+.. code-block:: yaml
+
+    # behat.yml
+    default:
+        config: # these are the default value
```
Comment (acoulton) [inline user_guide/configuration.rst]:
```suggestion
        config: # these are the default values
```

File: user_guide/configuration.rst [hunk 2]
```diff
@@ -111,6 +111,40 @@ All files from the ``imports`` block will be loaded by Behat and merged, in the
...
+        config:
+            stop_on_failure: true
+            strict: false
+
+    ci:
+        config:
+            stop_on_failure: false
+            strict: true
+
+This way, behat will stop on default environment and won't be strict, but will not stop and will be strict on CI.
```
Comment (acoulton) [inline user_guide/configuration.rst]:
```suggestion
This way, with the default profile behat will stop on failure and won't be strict, but will not stop and will be strict if the CI profile is selected.
```

**Review by acoulton — CHANGES_REQUESTED** [review]
Couple of tiny wording suggestions, otherwise this looks good and ready to merge once the feature PRs merge.

**Comment by jdeniau** [comment]
I convert this to "draft" until https://github.com/Behat/Behat/pull/1516#issuecomment-2456532362 is resolved.

**Comment by jdeniau** [comment]
> I convert this to "draft" until [Behat/Behat#1516 (comment)](https://github.com/Behat/Behat/pull/1516#issuecomment-2456532362) is resolved.

Mark as ready now that 3.16 is out.

File: user_guide/configuration.rst [hunk 3]
```diff
@@ -111,6 +111,40 @@ All files from the ``imports`` block will be loaded by Behat and merged, in the
...
+    ci:
+        testers:
+            stop_on_failure: false
+            strict: true
+
+This way, with the default profile behat will stop on failure and won't be strict, but will not stop and will be strict if the CI profile is selected.
+
+You can force `--stop-on-failure` and `--strict` via CLI options to override configuration values.
+
+
```
Comment (carlos-granados) [inline user_guide/configuration.rst]: I think you should remove this extra blank line

**Review by carlos-granados — CHANGES_REQUESTED** [review]
Small change needed

**Review by carlos-granados — APPROVED** [review]
All good, thanks @jdeniau

**Review by acoulton — APPROVED** [review]

**Comment by acoulton** [comment]
Great, thanks @jdeniau !

**Pull request merged**